### PR TITLE
fix(core): should validate config when creating a new connector instance

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

should validate config when creating a new connector instance

## Testing

UTs and ITs.

### CODE REVIEW

File: packages/shared/package.json [hunk 1]
```diff
@@ -32,6 +32,7 @@
     "test:ci": "pnpm test:only"
   },
   "devDependencies": {
+    "@logto/connector-kit": "workspace:*",
```
Comment (charIeszhao) [inline packages/shared/package.json:35]: Is this necessary?

Comment (darcyYe) [reply packages/shared/package.json:35]: We can not mock @logto/connector-kit methods with out this dependency. So seems it's necessary.

File: packages/core/src/libraries/connector.ts [hunk 2]
```diff
@@ -39,16 +39,11 @@ export const createConnectorLibrary = (queries: Queries) => {
...
-          const rawConnector = await createConnector({
-            getConfig: async () => {
-              return getConnectorConfig(id);
-            },
-          });
-          validateConnectorModule(rawConnector);
-          const rawMetadata = await parseMetadata(rawConnector.metadata, packagePath);
+          const { rawConnector, rawMetadata } = await buildRawConnector(
+            connectorFactory,
+            async (_id: string) => getConnectorConfig(id)
```
Comment (wangsijie) [inline packages/core/src/libraries/connector.ts:45]: the `_id` is useless

File: packages/core/src/routes/connector.test.ts [hunk 3]
```diff
@@ -190,6 +205,33 @@ describe('connector route', () => {
       );
     });
 
+    it('throw when post a new connector record with wrong config', async () => {
```
Comment (wangsijie) [inline packages/core/src/routes/connector.test.ts:208]:
```suggestion
    it('throw when create a new connector record with wrong config', async () => {
```

**Review by wangsijie — APPROVED** [review]
Overall LGTM

File: packages/core/src/routes/connector.test.ts (hunk 3 above)
Comment (darcyYe) [reply packages/core/src/routes/connector.test.ts]: Updated.

File: packages/core/src/libraries/connector.ts (hunk 2 above)
Comment (darcyYe) [reply packages/core/src/libraries/connector.ts]: removed.

**Pull request merged**